### ISSUE TITLE
:sort u merges two duplicates

### ISSUE THREAD

**Issue by karizma** [VSCodeVim/Vim#6825]
**Describe the bug**
I found some cases where when you use sort u, it sort of merges two duplicates

**To Reproduce**

1.  Open a file
2.  Input the following content: 
```
there
do
there
```
3.  Use `:sort u`

Output:
```
do
thereere
```
**Expected behavior**
```
do
there
```

**Environment**

- Extension (VsCodeVim) version: 1.21.4
- VSCode version: 1.57.1
- OS: Windows 10

---

Doing a bit more testing, it seems like it works with any 3 lines as long as the first and last are the same, and the middle is 'do':

```
candy
do
candy

=>

candy
dondy
```
```
hello
do
hello

=>

do
hellollo
```

**Comment by teachermugabo** [comment]
This is a fascinating error. I had no idea Vim had this kind of functionality. Looking forward to seeing how this gets resolved and to using it!

**Comment by karizma** [comment]
For the most part you can already use sort, it's pretty stable. Vscode itself also has the sort functionality in case you didn't know (although it's way more limited than vim).

**Comment by karizma** [comment]
Not sure if you wanted to get ping, but it has been solved at #6835. @teachermugabo

**Comment by teachermugabo** [comment]
Thank you!
